Index Benchmark with TPCH

hjhhsy120 opened the pull request.

amlatyr committed:
Adding Vectorization...

amlatyr committed:
Partial Commit. I am working on the compiler_test.cpp, the tests are interacting with one another. So run tests in compiler_test one by one (comment out the others)

amlatyr committed:
Adding vectorization...

hjhhsy120 committed:
Add Benchmark for BwTree Index

hjhhsy120 committed:
add readme for index benchmark

hjhhsy120 committed:
add some comments

amlatyr committed:
Working on vectorization. Will split vectorization to another branch

amlatyr committed:
Separate vectorization

hjhhsy120 committed:
modified benchmark for index

hjhhsy120 committed:
Update README.md

hjhhsy120 committed:
modified cmake file for jemalloc

hjhhsy120 committed:
Merge branch 'master' of https://github.com/hjhhsy120/terrier

hjhhsy120 committed:
Merge remote-tracking branch 'amadou/tpl-without-codegen'

hjhhsy120 committed:
Merge remote-tracking branch 'amadou/tpl-without-codegen'

hjhhsy120 committed:
Add files for index with tpch

hjhhsy120 committed:
Index with tpch

hjhhsy120 committed:
Index with tpch

hjhhsy120 committed:
Index with tpch

hjhhsy120 committed:
Index with tpch

hjhhsy120 committed:
Index with tpch debugging

hjhhsy120 committed:
8.7

hjhhsy120 committed:
Merge remote-tracking branch 'amadou/tpl-without-codegen'

hjhhsy120 committed:
Merge 8.7

hjhhsy120 committed:
Merge remote-tracking branch 'amadou/tpl-without-codegen'

hjhhsy120 committed:
Index with tpch

hjhhsy120 committed:
debugging

hjhhsy120 committed:
debugging

hjhhsy120 committed:
debugging

hjhhsy120 committed:
fixed bugs

hjhhsy120 committed:
Turn off loggers

hjhhsy120 committed:
Index with tpch

hjhhsy120 committed:
Add switches for empty, loop and array tests

hjhhsy120 committed:
Fixed loop test

hjhhsy120 committed:
Fix loop task

hjhhsy120 committed:
Fix core for gc

hjhhsy120 committed:
Abandon the last exec_ms

hjhhsy120 committed:
Disable adaptive and record time separately

hjhhsy120 committed:
Modified index with tpch

hjhhsy120 committed:
Add timer for insertion

hjhhsy120 committed:
Add small scan buffer

amlatyr committed:
Merge tpl-without-codegen before remove csv table generation

hjhhsy120 committed:
Debugging?

hjhhsy120 committed:
Add 10M array test

hjhhsy120 committed:
Add multiple tpch benchmark

hjhhsy120 committed:
Compile once and fix tpch file

amlatyr committed:
Merge branch 'tpl-without-codegen' into tpl-with-codegen

amlatyr committed:
Updating branch...

amlatyr committed:
Merge branch 'tpl-without-codegen' into tpl-with-codegen

amlatyr committed:
Updating compiler. Need to pass in parent and child operators....

hjhhsy120 committed:
Tpch with cache

amlatyr committed:
Update code to reflect new APIs

amlatyr committed:
Merge branch 'tpl-without-codegen' into tpl-with-codegen

hjhhsy120 committed:
Read file once

hjhhsy120 committed:
fix bug

hjhhsy120 committed:
Fix vector bug

lmwnshn committed:
Bump CMake minimum version from 3.2 to 3.8, CMAKE_CXX_STANDARD 17 is only valid starting with CMake 3.8. (#521)

hjhhsy120 committed:
Add ONE_TEST

hjhhsy120 committed:
prepare for big experiment

mbutrovich committed:
Add readability-identifier-naming options to .clang-tidy file (#524)

tli2 committed:
Refactor deferred action framework (#441)

linmagit commented on src/include/execution/tplclass.h:
```diff
@@ -0,0 +1,422 @@
+
+        static const int max_num_modules_ = 6;
+        std::map <std::string, int> name_id_;
+        std::unique_ptr <vm::Module> modules_[max_num_modules_];
```
Can't we use a single map to directly map `string` to `Module`?

linmagit commented on src/include/execution/tplclass.h:
```diff
@@ -0,0 +1,422 @@
+                    }
+                    EXECUTION_LOG_INFO("ADAPTIVE main() returned: {}", main());
+                }
+            }
```
Add another flag as a member variable to select which mode to run, instead of leaving commented code in the file.

linmagit commented on src/include/execution/tplclass.h:
```diff
@@ -0,0 +1,422 @@
+
+                CompileAndRun(input);
+            }
+        }*/
```
Remove unused code.

linmagit commented on src/include/execution/tplclass.h:
```diff
@@ -0,0 +1,422 @@
+ * @param name The name of the module/program
+ */
+        //void CompileAndRun(const std::string &source, const std::string &name = "tmp-tpl") {
+
```
Create a separate cpp file to put all the functions below there:
https://github.com/cmu-db/terrier/wiki/Coding-Guideline#h-and-cpp-files

And then use forward declaration to remove as much header includes as possible:
https://github.com/cmu-db/terrier/wiki/Coding-Guideline#forward-declaration

linmagit commented on src/include/execution/tplclass.h:
```diff
@@ -0,0 +1,422 @@
+        terrier::catalog::Catalog * catalog_pointer_;
+        bool *unfinished_;
+
+        static const int max_num_modules_ = 6;
```
Can we make this non-static?

linmagit commented on benchmark/index/multiple_tpch.cpp:
```diff
@@ -0,0 +1,294 @@
+// Whether it is a local test with small numbers
```
A few high-level things for the first pass:
1. Please replace all these macros to member variables in the class. Part of the coding convention in our system is not to use macros if possible. Macros reduce the readability of the code, and also introduce global states.
2. Please make all the member variables non-static (unless necessary). It's better to isolate as much information in the class instance itself as possible.
3. Do not hard-code the size of arrays (like `tpch_filename_` or `interp_exec_ms_sum_`). Use vectors instead of make them be able to handle variable sizes without the need to recompile the program. You can initialize them in the `SetUp` function if you want.
4. Break the `RandomInsert` benchmark function into smaller blocks or sub-functions:
https://github.com/cmu-db/terrier/wiki/Coding-Guideline#the-wall-of-text
5. Remove unused code.

linmagit reviewed (changes requested):
Took a first pass of the code. Please see the detailed comments.

amlatyr committed:
bla bla bla

amlatyr committed:
Merge remote-tracking branch 'upstream/master' into tpl-without-codegen

amlatyr committed:
Run format

hjhhsy120 committed:
Reconstruction

hjhhsy120 committed:
Add switch for single test

hjhhsy120 committed:
Fixed some bugs

hjhhsy120 committed:
Small modifications

hjhhsy120 committed:
Comments and small modifications

hjhhsy120 committed:
fixing a bug

hjhhsy120 committed:
Add TPCH benchmark and fixing bugs

hjhhsy120 committed:
fix a bug

apavlo committed:
Statistics Utilities (#528)

Optimizer Stats Utility Code:
* Histogram
* TopKElements
* CountMinSketch
* Table + Column Stats Objects

amlatyr committed:
Fixing clang-tidy. About to run -fix-errors....

hjhhsy120 committed:
Add a few flags

hjhhsy120 committed:
Changed table size

hjhhsy120 committed:
fixed a bug

hjhhsy120 committed:
fixed a bug

tli2 committed:
remove dead code (#532)

hjhhsy120 committed:
different tables for tpch as benchmark and other workloads

amlatyr committed:
Fix clang-tidy. Fixing JIT...

hjhhsy120 committed:
fix some bugs

amlatyr committed:
Get all checks to pass and fix JIT

amlatyr committed:
Fix apple code

amlatyr committed:
Merge remote-tracking branch 'remotes/upstream/master' into tpl-without-codegen

amlatyr committed:
revert benchmark changes

mbutrovich committed:
Use numactl instead of taskset for run_micro_bench.py (#529)

amlatyr committed:
Merge remote-tracking branch 'upstream/master' into tpl-without-codegen

amlatyr committed:
Fix build errors

amlatyr committed:
Move macros

amlatyr committed:
Format

amlatyr committed:
Fix mac visibility

amlatyr committed:
Temp Revert CMake version

amlatyr committed:
Fix compilation

amlatyr committed:
Remove visibility hidden from mac

apavlo committed:
Fixes for CountMinSketch tests. This greatly reduces the probability of a spurious failure. (#533)

* We reduce the # of heavy hitters and use exponentially larger numbers between them.
* Increase the width of sketches that are more sensitive to differences.

This takes care of #530

mbutrovich committed:
clang-tidy: performance-unnecessary-value-param (#535)

amlatyr committed:
Print output of TPL tests

amlatyr committed:
Format

amlatyr committed:
Merge remote-tracking branch 'upstream/master' into tpl-without-codegen

jrolli committed:
Make Jenkins print hostname, increase build parallelism on Jenkins Linux machines. (#525)

Commit:
Insert contention problems (#516)

tli2 committed:
GC queue and Deferred Action queue Ordering (#527)

amlatyr committed:
Attempt Apple fix

amlatyr committed:
Fix visibility issues by adding an explicit export to TPL functions and classes

amlatyr committed:
Run Format

amlatyr committed:
Simplify build. Write more tests

amlatyr committed:
Fix date comparison

amlatyr committed:
Reduce check-tpl time

amlatyr committed:
Bring back tpch reader

amlatyr committed:
Merge branch 'tpl-without-codegen' into tpl-with-codegen

amlatyr committed:
Update API

mbutrovich committed:
Update lookback. (#542)

mbutrovich committed:
Change catalog's #define'd constants to be constexpr instead (#538)

amlatyr committed:
Add PR type. Updated IndexIterator to use it

hjhhsy120 committed:
add ULOOP and USCAN, changed table generation sequence to debug

hjhhsy120 committed:
debug

amlatyrngom committed:
Tpl without codegen (#398)

This is an amazing PR that brings in the TPL execution engine without codege. There are way too many things to mention here. This is a massive commit.

Special thanks to @amlatyrngom + @pmenon for making this happen! #398

hjhhsy120 committed:
debug

mbutrovich committed:
Basic catalog microbenchmarks (#545)

mbutrovich committed:
Catalog: reuse ProjectedRowInitializers and maps (#546)

hjhhsy120 committed:
debug

hjhhsy120 committed:
change flag

Cabel496 committed:
Add helper function to extract referenced column oids from AbstractExpression (#539)

GustavoAngulo committed:
Better GetIndexes method (#553)

tanujnay112 committed:
Added Index OID vector to InsertPlanNode (#543)

GustavoAngulo committed:
Recovery Manager (#459)

mbutrovich committed:
Hash index based on libcuckoo (#485)

amlatyr committed:
Add tuple slot type. Update index test

hjhhsy120 committed:
add UTABLE

mbutrovich committed:
Update LLVM_DIR path for llvm@8 Homebrew package (#559)

mbutrovich committed:
Move coverage to Jenkins from Travis (#551)

GustavoAngulo committed:
Add recovery benchmark to nightly runs and new index recovery benchmark (#558)

Commit:
use nproc instead of hardcoded number for number of make jobs (#562)

mbutrovich committed:
Multiplatform thread CPU timer (#563)

apavlo committed:
Add missing documentation for typedefs (#570)

* Added missing documentation for two typedefs that Doxygen on Travis occassionally complains about #569

* Updated .gitignore to hide libpqxx files in root dir

mbutrovich committed:
Simplified ThreadCPUTimer test. (#573)

apavlo committed:
AstDump Unit Test (#572)

amlatyr committed:
Merge branch 'tpl-tpcc' into tpl-with-codegen

hjhhsy120 committed:
before merging remote-tracking branch 'cmudb/master'

hjhhsy120 committed:
add xxhash license

hjhhsy120 committed:
Merge remote-tracking branch 'amadou/tpl-with-codegen'

hjhhsy120 committed:
fix some bugs

amlatyr committed:
Use Projected rows and add Index Scan

amlatyr committed:
Uncomment tests

amlatyr committed:
Refactor expression generation. Add PR filling.

amlatyr committed:
Add comments

hjhhsy120 committed:
debugging

hjhhsy120 committed:
Merge remote-tracking branch 'amadou/tpl-with-codegen'
